build.sh

```diff
@@ -32,60 +32,76 @@ region = $HEROKLES_AWS_REGION
 eocre
 
 echo "Getting environment variables."
-JSON=
-JSON_FULL=$( aws ssm get-parameters --name /${PROJECT}/${ENV} )
-
-if [[ ! -z $( echo "$JSON_FULL" | jq -r '.InvalidParameters | .[]' ) ]] ; then
-  if [[ $ENV == pr-${PR_NUM:-''} ]] ; then
-    echo "New PR deployment, copying env vars from the template."
-    JSON=$( aws ssm get-parameters --name /${PROJECT}/prs | jq -r '.Parameters | .[] | .Value' )
+JSON=$( mktemp )
+JSON_TEMPLATE=$( mktemp )
+JSON_FULL=$( mktemp )
+aws ssm get-parameters --name /${PROJECT}/${ENV} > $JSON_FULL
+
+if [[ $ENV == pr-${PR_NUM:-''} ]] ; then
+  aws ssm get-parameters --name /${PROJECT}/prs > $JSON_TEMPLATE # get all template envs
+  if [[ ! -z $( jq -r '.InvalidParameters | .[]' $JSON_TEMPLATE ) ]] ; then
+    echo "Template PR variables /${PROJECT}/prs not found."
+    exit 1
+  fi
+  if [[ ! -z $( jq -r '.InvalidParameters | .[]' $JSON_FULL ) ]] ; then # check if it's a new PR
+    echo "New PR - create variables."
+    jq -r '.Parameters | .[] | .Value' $JSON_TEMPLATE > $JSON
     if [[ -f herokles/set-custom-pr-envs.sh ]] ; then
-      source ./herokles/set-custom-pr-envs.sh
+      ./herokles/set-custom-pr-envs.sh $JSON
```

Comment:
This requires the script to append to json already in a file? From scripts author perspective it may be easier to output dot env format
```
A=B
C=D
```
That can be taken and converted to json here. If devs would be authoring custom pr envs it would be easier :shrug:

Reply:
Yeah, I don't like that solution particularly either. But it's just so much more convenient to work with json files rather than json in an env var. How would the `A=B` setting work otherwise though? It would either had to be `echoed`, or at least written to _some_ file, because I can't imagine "sourcing" the env vars any other way.

Reply:
It would be echoed and caught in here. Then it would be parsed and turned into json here.